src/SFA.DAS.EAS.Web/Orchestrators/EmployerAgreementOrchestrator.cs

```diff
@@ -297,5 +301,141 @@ public EmployerAgreementOrchestrator(IMediator mediator, ILogger logger, Employe
+                    Severity = FlashMessageSeverityLevel.Error
+                };
+                response.Exception = ex;
+            }
+            catch (UnauthorizedAccessException ex)
+            {
+                response.Status = HttpStatusCode.Unauthorized;
+                response.Exception = ex;
+            }
+            return response;
+        }
+
+        public virtual async Task<OrchestratorResponse<ConfirmLegalAgreementToRemoveViewModel>> GetConfirmRemoveOrganisationViewModel(string agreementId, string hashedAccountId, string userId)
+        {
+            var response = new OrchestratorResponse<ConfirmLegalAgreementToRemoveViewModel>();
+            try
+            {
+                var result = await _mediator.SendAsync(new GetAccountEmployerAgreementRemoveRequest
+                {
+                    HashedAccountId = hashedAccountId,
+                    UserId = userId,
+                    HashedAgreementId = agreementId
+                });
+                response.Data = new ConfirmLegalAgreementToRemoveViewModel
+                {
+                    HashedAccountId = result.Agreement.HashedAccountId,
+                    HashedAgreementId = result.Agreement.HashedAgreementId,
+                    Id = result.Agreement.Id,
+                    Name = result.Agreement.Name
+                };
+            }
+            catch (InvalidRequestException ex)
+            {
+                response.Status = HttpStatusCode.BadRequest;
+                response.FlashMessage = new FlashMessageViewModel
+                {
+                    Headline = "Errors to fix",
+                    Message = "Check the following details:",
+                    ErrorMessages = ex.ErrorMessages,
+                    Severity = FlashMessageSeverityLevel.Error
+                };
+                response.Exception = ex;
+            }
+            catch (UnauthorizedAccessException ex)
+            {
+                response.Status = HttpStatusCode.Unauthorized;
+                response.Exception = ex;
+            }
+
+            return response;
+        }
+
+        public virtual async Task<OrchestratorResponse<bool>>  RemoveLegalAgreement(ConfirmLegalAgreementToRemoveViewModel model, string userId)
+        {
+            var response = new OrchestratorResponse<bool>();
+
+            try
+            {
+                if (model.RemoveOrganisation == null)
+                {
+                    response.Status = HttpStatusCode.BadRequest;
+                    response.FlashMessage = new FlashMessageViewModel
+                    {
+                        Headline = "Errors to fix",
+                        Message = "Check the following details:",
+                        ErrorMessages = new Dictionary<string, string> { { "RemoveOrganisation", "Confirm you wish to remove the organisation" } },
+                        Severity = FlashMessageSeverityLevel.Error
+                    };
```

Comment:
Could the FlashMessage creation be extracted out into a function to save all the duplication of strings and condense it down to a 1 line function call with a single parameter?